modules/imgcodecs/src/loadsave.cpp

```diff
@@ -1242,6 +1242,13 @@ void ImageCollection::Impl::init(String const& filename, int flags) {
     m_decoder->setSource(filename);
     CV_Assert(m_decoder->readHeader());
 
+    m_size = m_decoder->getFrameCount();
+
+    if (m_size > 0) {
+        m_pages.resize(m_size);
+        return;
+    }
+
     // count the pages of the image collection
     size_t count = 1;
     while(m_decoder->nextPage()) count++;
```

Comment:
It should be removed then, right? As soon as following `m_size = count;`

Reply:
Also there is GDAL branch bellow.

Reply:
```
    m_size = m_decoder->getFrameCount();

    if (m_size > 0) {
        m_pages.resize(m_size);
        return;
    }
```
if `m_frame_count` set by `m_decoder->readHeader()` set m_size and return.. otherways the following code needed.

Reply:
let me look GDAL decoder.

Reply:
`m_frame_count = 1;` default for all decoders except AvifDecoder and TiffDecoder.